backend/kernelCI_app/management/commands/notifications.py

```diff
@@ -455,12 +458,39 @@ def process_submission_options(
     return recipients
 
 
+def discard_sent_reports(
+    tree_key_set: set[TreeKey],
+    tree_prop_map: dict[TreeKey, list[dict[str, Any]]],
+    skip_sent_reports: bool,
+) -> tuple[set[TreeKey], dict[TreeKey, list[dict[str, Any]]]]:
+    if skip_sent_reports:
+        return tree_key_set, tree_prop_map
+
+    key_set_copy = tree_key_set.copy()
+    prop_map_copy = tree_prop_map.copy()
+
+    already_sent_report_checkouts = check_sent_notifications(prop_map_copy)
+    discarded = 0
+    for sent_checkout in already_sent_report_checkouts:
+        key_set_copy.discard(sent_checkout)
+        prop_map_copy.pop(sent_checkout, None)
+        discarded += 1
+    print(
+        "Discarded %d reports already sent in the past %s"
+        % (discarded, RESEND_INTERVAL)
+    )
```

Comment:
Can we replace for `len(already_sent_report_checkouts)`?

```suggestion
    for sent_checkout in already_sent_report_checkouts:
        key_set_copy.discard(sent_checkout)
        prop_map_copy.pop(sent_checkout, None)
    print(
        "Discarded %d reports already sent in the past %s"
        % (len(already_sent_report_checkouts), RESEND_INTERVAL)
    )
```